src/docfx/validation/CustomRuleProvider.cs

```diff
@@ -39,18 +39,21 @@ internal class CustomRuleProvider
 
         public bool IsEnable(FilePath filePath, CustomRule customRule, string? moniker = null)
         {
-            var canonicalVersion = _publishUrlMap.Value.GetCanonicalVersion(filePath);
-
             // If content versioning not enabled for this depot, canonicalVersion will be null, content will always be the canonical version;
             // If content versioning enabled and moniker is null, we should check file-level monikers to be sure;
             // If content versioning enabled and moniker is not null, just compare canonicalVersion and moniker.
-            var isCanonicalVersion = string.IsNullOrEmpty(canonicalVersion) ? true :
-                string.IsNullOrEmpty(moniker) ? _monikerProvider.GetFileLevelMonikers(_errorLog, filePath).IsCanonicalVersion(canonicalVersion) :
-                canonicalVersion == moniker;
-
-            if (customRule.CanonicalVersionOnly && !isCanonicalVersion)
+            // for test case in toc.yml, publishUrlMap is null
```

Comment:
hmm, it is strange that `_publishUrlMap.Value` is null, the code before this change calls the `GetCanonicalVersion` method immediately?

Reply:
As talked offline, revert cs changes.